### PR TITLE
[custom channels]: Assert fix for duplicate script key issues in MPP payments

### PR DESCRIPTION
**Pull request by guggero**
Updates the force close HTLC sweep integration test to also do MPP payments, to make sure duplicate script keys due to identical MPP shard HTLCs don't cause an issue.

### CODE REVIEW

**Review by GeorgeTsagk — APPROVED** [review]
Lgtm!

File: itest/litd_custom_channels_test.go [hunk 1]
```diff
@@ -2870,7 +2880,7 @@ func testCustomChannelsHtlcForceClose(_ context.Context, net *NetworkHarness,
 	var (
 		bobHodlInvoices   []assetHodlInvoice
 		aliceHodlInvoices []assetHodlInvoice
-		assetInvoiceAmt   = 100
+		assetInvoiceAmt   = 5_000
```
Comment (jharveyb) [inline itest/litd_custom_channels_test.go:2883]: nit: Bit confused on the math for this change causing MPP to get used.

AFAICT, setting `mpp = true` below limits the shard size to 80e6 msats / 80000 sats.

But the minted asset seems to be 1,000,000 `itest-asset-cents`. And it looks like `assetInvoiceAmt` is denominated in cents also. Not sure if decimal display is being set somewhere I can't find.

I also can't find the exchange rate being used here that would probably help clarify this math.

Comment (jharveyb) [reply itest/litd_custom_channels_test.go:2883]: Update: I found the rate, from grepping after inspecting the logs, as the config value of `mockoracleassetsperbtc`. Would still be good to call out.

Also this log msg should be `got quote for %d msats`, not sats:

https://github.com/lightninglabs/lightning-terminal/blob/a72e388e5f5d3540d1c90e8b28664bbb7b1c9708/itest/assets_test.go#L1129

So, the quoted sat amount is ~85901 sats, which is over the MPP shard threshold of 80000 sats / 4656 asset units.

**Review by jharveyb — APPROVED** [review]
LGTM - confirmed MPP behavior from peeking at the logs 💯

File: itest/litd_custom_channels_test.go [hunk 2]
```diff
@@ -2903,9 +2913,14 @@ func testCustomChannelsHtlcForceClose(_ context.Context, net *NetworkHarness,
...
 	}
 
-	// At this point, both sides should have 4 HTLCs active.
-	assertNumHtlcs(t.t, alice, 4)
-	assertNumHtlcs(t.t, bob, 4)
+	// At this point, both sides should have 4 (or +2 with MPP) HTLCs
+	// active.
+	numHtlcs := 4
+	if mpp {
+		numHtlcs += 2
```
[the remark this answers is not in the document]
Comment (guggero) [reply itest/litd_custom_channels_test.go:2920]: Right, makes sense. That way we can also more easily increase the number of HTLCs later on.

File: itest/litd_custom_channels_test.go (hunk 1 above)
Comment (guggero) [reply itest/litd_custom_channels_test.go:2883]: Ah, yes, that's not super apparent. Added a comment to describe that, which then also makes the use of the `mpp` variable a bit more apparent.

File: go.mod [hunk 3]
```diff
@@ -16,14 +16,14 @@ require (
...
-	github.com/lightninglabs/lndclient v0.18.4-6
+	github.com/lightninglabs/lndclient v0.18.4-7
 	github.com/lightninglabs/loop v0.28.8-beta.0.20241022072406-1e8ae31ddc27
 	github.com/lightninglabs/loop/looprpc v1.0.0
 	github.com/lightninglabs/loop/swapserverrpc v1.0.10
 	github.com/lightninglabs/pool v0.6.5-beta.0.20241015105339-044cb451b5df
 	github.com/lightninglabs/pool/auctioneerrpc v1.1.2
 	github.com/lightninglabs/pool/poolrpc v1.0.0
-	github.com/lightninglabs/taproot-assets v0.4.2-0.20241125202051-783fb1e5ab03
+	github.com/lightninglabs/taproot-assets v0.4.2-0.20241127150634-c96f36961231
```
Comment (gijswijs) [inline go.mod:26]: Could bump it now to RC5, right?

Comment (guggero) [reply go.mod:26]: Yes, will do that once the tag is pushed in `tapd`.

File: itest/litd_custom_channels_test.go (hunk 1 above)
Comment (gijswijs) [reply itest/litd_custom_channels_test.go]: I didn't grok this either. Thanks for explaining it @jharveyb.

**Pull request merged**